apps/studio/electron/main/code/classes.ts

```diff
@@ -35,5 +35,23 @@ function getNodeClasses(node: t.JSXElement): string[] {
         return classNameAttr.value.expression.value.split(/\s+/).filter(Boolean);
     }
 
+    if (
```

Comment:
We could also return an error straight from here since we're only handling classes. Everything else passes through.

Reply:
I reformatted the code into:

```typescript
    if (
        t.isJSXExpressionContainer(classNameAttr.value) &&
        t.isTemplateLiteral(classNameAttr.value.expression)
    ) {
        const templateLiteral = classNameAttr.value.expression;

        // Immediately return error if dynamic classes are detected within the template literal
        if (templateLiteral.expressions.length > 0) {
            return {
                type: 'error',
                reason: 'Dynamic classes detected. Dynamic variables in the className prevent extraction of Tailwind classes.',
            };
        }

        // Extract and return static classes from the template literal if no dynamic classes are used
        const quasis = templateLiteral.quasis.map((quasi) => quasi.value.raw.split(/\s+/));
        return {
            type: 'classes',
            value: quasis.flat().filter(Boolean),
        };
    }
```

This allows the code to exit the block early if it is an error (dynamic classes are used).
I am unsure if this is exactly what you meant. Can you please clarify further if I misinterpreted it? :)

Reply:
This works great, thanks!